File: SonicXBackendC/SonicXBackendC/PlaylistManagment/PlaylistDAO.cs
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using SonicXBackendC.DataBase;
using SonicXBackendC.MusicManagment;
using SonicXBackendC.UserManagement;
using MySql.Data.MySqlClient;

namespace SonicXBackendC.PlaylistManagment
{
   
    public class PlaylistDAO
    {
        private readonly MySqlConnection connection;

        public PlaylistDAO()
        {
            try
            {
                this.connection = DatabaseConnector.GetInstance().GetConnection();
            }
            catch (Exception e)
            {
                throw new Exception("Veritabanı bağlantısı oluşturulurken hata oluştu: " + e.Message);
            }
        }

        public void AddMusicFileToPlaylist(string playlistName, MusicFile musicFile, UserService userService)
        {
            string sql = "INSERT INTO playlist_" + userService.GetUser().Id + " (playlist_name, title, artist, album, duration, path) VALUES (@PlaylistName, @Title, @Artist, @Album, @Duration, @Path)";
            try
            {
                using (MySqlCommand command = new MySqlCommand(sql, connection))
                {
                    command.Parameters.AddWithValue("@PlaylistName", playlistName);
                    command.Parameters.AddWithValue("@Title", musicFile.Title);
                    command.Parameters.AddWithValue("@Artist", musicFile.Artist);
                    command.Parameters.AddWithValue("@Album", musicFile.Album);
                    command.Parameters.AddWithValue("@Duration", musicFile.Duration);
                    command.Parameters.AddWithValue("@Path", musicFile.Path);
                    command.ExecuteNonQuery();
                    Console.WriteLine("Müzik dosyası çalma listesine eklendi: " + musicFile.Title);
                }
            }
            catch (SqlException e)
            {
                Console.WriteLine("Müzik dosyası çalma listesine eklenirken hata oluştu: " + e.Message);
            }
        }

        public void RemoveMusicFileFromPlaylist(string playlistName, string musicTitle, UserService userService)
        {
            string sql = "DELETE FROM playlist_files WHERE playlist_name = @PlaylistName AND title = @Title AND user_id = @UserId";
            try
            {
                using (MySqlCommand command = new MySqlCommand(sql, connection))
                {
                    command.Parameters.AddWithValue("@PlaylistName", playlistName);
                    command.Parameters.AddWithValue("@Title", musicTitle);
                    command.Parameters.AddWithValue("@UserId", userService.GetUser().Id);
                    int rowsDeleted = command.ExecuteNonQuery();
                    if (rowsDeleted > 0)
                    {
                        Console.WriteLine("Müzik dosyası çalma listesinden silindi: " + musicTitle);
                    }
                    else
                    {
                        Console.WriteLine("Belirtilen müzik dosyası çalma listesinde bulunamadı: " + musicTitle);
                    }
                }
            }
            catch (SqlException e)
            {
                Console.WriteLine("Müzik dosyası çalma listesinden silinirken hata oluştu: " + e.Message);
            }
        }

        public List<MusicFile> GetMusicFilesInPlaylist(string playlistName, UserService userService)
        {
            List<MusicFile> musicFiles = new List<MusicFile>();
            string sql = "SELECT title, artist, album, duration, path FROM playlist_" + userService.GetUser().Id + " WHERE playlist_name = @PlaylistName";
            try
            {
                using (MySqlCommand command = new MySqlCommand(sql, connection))
                {
                    command.Parameters.AddWithValue("@PlaylistName", playlistName);
                    using (MySqlDataReader reader = command.ExecuteReader())
                    {
                        while (reader.Read())
                        {
                            string title = reader.GetString(reader.GetOrdinal("title"));
                            string artist = reader.GetString(reader.GetOrdinal("artist"));
                            string album = reader.GetString(reader.GetOrdinal("album"));
                            int duration = reader.GetInt32(reader.GetOrdinal("duration"));
                            string path = reader.GetString(reader.GetOrdinal("path"));
                            MusicFile musicFile = new MusicFile(title, artist, album, duration, path);
                            musicFiles.Add(musicFile);
                        }
                    }
                }
            }
            catch (MySqlException e)
            {
                Console.WriteLine("Çalma listesi yenilenirken hata oluştu: " + e.Message);
            }
            return musicFiles;
        }

        public List<string> GetAllPlaylistNames(UserService userService)
        {
            List<string> playlistNames = new List<string>();
            string sql = "SELECT DISTINCT playlist_name FROM playlist_" + userService.GetUser().Id;
            try
            {
                using (MySqlCommand command = new MySqlCommand(sql, connection))
                {
                    using (MySqlDataReader reader = command.ExecuteReader())
                    {
                        while (reader.Read())
                        {
                            string playlistName = reader.GetString(reader.GetOrdinal("playlist_name"));
                            playlistNames.Add(playlistName);
                        }
                    }
                }
            }
            catch (Exception e)
            {
                Console.WriteLine("Çalma listeleri getirilirken hata oluştu: " + e.Message);
            }
            return playlistNames;
        }

        public void Close()
        {
            try
            {
                DatabaseConnector.GetInstance().Disconnect();
            }
            catch (SqlException e)
            {
                throw new Exception("Veritabanı bağlantısı kapatılırken hata oluştu: " + e.Message);
            }
        }
    }
}

File: SonicXBackned/src/main/java/org/example/PlaylistManagment/PlaylistDAO.java
package org.example.PlaylistManagment;

import org.example.DataBase.DatabaseConnector;
import org.example.MusicManagment.MusicFile;
import org.example.UserManagement.User;
import org.example.UserManagment.UserService;

import javax.management.RuntimeErrorException;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class PlaylistDAO {

    private final Connection connection;

    public PlaylistDAO() throws SQLException {
        this.connection = DatabaseConnector.getInstance().getConnection();
    }

    public void addMusicFileToPlaylist(String playlistName, MusicFile musicFile, UserService userService) throws SQLException {
        String sql = "INSERT INTO playlist_"+userService.getUser().getId() +"(playlist_name, title, artist, album, duration, path) VALUES (?, ?, ?, ?, ?, ?)";
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setString(1, playlistName);
            statement.setString(2, musicFile.getTitle());
            statement.setString(3, musicFile.getArtist());
            statement.setString(4, musicFile.getAlbum());
            statement.setInt(5, musicFile.getDuration());
            statement.setString(6, musicFile.getPath());
            statement.executeUpdate();
            System.out.println("Müzik dosyası çalma listesine eklendi: " + musicFile.getTitle());
        }
    }

    public void removeMusicFileFromPlaylist(String playlistName, String musicTitle, UserService userService) throws SQLException {
        String sql = "DELETE FROM playlist_files WHERE playlist_name = ? AND title = ? AND user_id = ?";
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setString(1, playlistName);
            statement.setString(2, musicTitle);
            statement.setInt(3, userService.getUser().getId());
            int rowsDeleted = statement.executeUpdate();
            if (rowsDeleted > 0) {
                System.out.println("Müzik dosyası çalma listesinden silindi: " + musicTitle);
            } else {
                System.out.println("Belirtilen müzik dosyası çalma listesinde bulunamadı: " + musicTitle);
            }
        }
    }

    public List<MusicFile> getMusicFilesInPlaylist(String playlistName, UserService userService) throws SQLException {
        List<MusicFile> musicFiles = new ArrayList<>();
        String sql = "SELECT title, artist, album, duration, path FROM playlist_"+ userService.getUser().getId()+" WHERE playlist_name = ?";
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setString(1, playlistName);
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                String title = resultSet.getString("title");
                String artist = resultSet.getString("artist");
                String album = resultSet.getString("album");
                int duration = resultSet.getInt("duration");
                String file=resultSet.getString("path");
                MusicFile musicFile = new MusicFile(title, artist, album, duration,file);
                musicFiles.add(musicFile);

            }
        } catch (SQLException e) {
            System.out.println("Çalma listesi yenilenirken hata oluştu: " + e.getMessage());
        }

        return musicFiles;
    }

    public List<String> getAllPlaylistNames(UserService userService) throws SQLException {
        List<String> playlistNames = new ArrayList<>();
        String sql = "SELECT DISTINCT playlist_name FROM playlist_" + userService.getUser().getId();
        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            ResultSet resultSet = statement.executeQuery();
            while (resultSet.next()) {
                String playlistName = resultSet.getString("playlist_name");
                playlistNames.add(playlistName);
            }
            resultSet.close();
        }catch (Exception e){
            System.out.println(e.getMessage());
        }
        return playlistNames;
    }

    public void close()  {
        try {
            DatabaseConnector.getInstance().disconnect();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
}
